roles/ceph-common/tasks/main.yml

```diff
@@ -81,6 +81,27 @@
     dest: "{{ cinder_uuid_file['source'] }}"
     content: "{{ cinder_uuid_file['content'] | b64decode }}"
 
+# generate uuid for nova.conf and secret1.xml
+- name: generate nova uuid
+  shell: uuidgen > /etc/ceph/nova_uuid && truncate -s -1 /etc/ceph/nova_uuid
+         creates=/etc/ceph/nova_uuid
+  run_once: true
+  delegate_to:  "{{ groups['ceph_monitors'][0] }}"
+  when: nova.ceph_backend|bool
+
+- name: fetch contents of nova_uuid file
+  slurp: path=/etc/ceph/nova_uuid
+  run_once: true
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  register: nova_uuid_file
+  when: nova.backend_ceph|bool
+  
+- name: copy nova_uuid to all hosts
+  copy:
+    dest: "{{ nova_uuid_file['source'] }}"
+    content: "{{ nova_uuid_file['content'] | b64decode }}"
+  when: nova.backend_ceph|bool
```

Comment:
All of these tasks can be put into a `block`, which will have one conditional on the entire block. Saves some writing. http://docs.ansible.com/ansible/playbooks_blocks.html